### PR TITLE
feature(planner): Migrate DROP USER statement to new planner

### LINKED ISSUE #5777: Migrate `USER` related DDL statements to new planner
## User related
- [x] `CREATE USER` https://github.com/datafuselabs/databend/pull/5802
- [x] `DROP USER` https://github.com/datafuselabs/databend/pull/5813
- [x] `ALTER USER`https://github.com/datafuselabs/databend/pull/5823
- [x] `SHOW USERS` https://github.com/datafuselabs/databend/pull/6016

## Role related
- [x] `CREATE ROLE` https://github.com/datafuselabs/databend/pull/5935
- [x] `DROP ROLE` https://github.com/datafuselabs/databend/pull/5935
- [x] `SHOW ROLES` https://github.com/datafuselabs/databend/pull/6016

## Privilege related
- [x] `GRANT <privileges>` https://github.com/datafuselabs/databend/pull/6049
- [x] `GRANT ROLE` https://github.com/datafuselabs/databend/pull/6049
- [ ] `REVOKE <privileges>` https://github.com/datafuselabs/databend/pull/6066
- [ ] `REVOKE ROLE` https://github.com/datafuselabs/databend/pull/6066
- [x] `SHOW GRANTS` https://github.com/datafuselabs/databend/pull/6049

### PR DESCRIPTION
**Pull request by TCeason**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Migrate drop user to new planner

## Changelog

- New Feature

## Related Issues

 #5777

### CODE REVIEW

**Review by leiysky — APPROVED** [review]

File: tests/suites/0_stateless/05_ddl/05_0007_ddl_drop_user_v2.sql [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+CREATE USER 'test-j'@'localhost' IDENTIFIED BY 'password';
+DROP USER 'test-j'@'localhost';
+DROP USER IF EXISTS 'test-j'@'localhost';
+
+CREATE USER 'test-l'@'localhost' IDENTIFIED WITH sha256_password BY 'password';
+DROP USER 'test-l'@'localhost';
+DROP USER IF EXISTS 'test-l'@'localhost';
+DROP USER IF EXISTS 'test-l'@'localhost';
+DROP USER IF EXISTS 'test-l'@'localhost';
```
Comment (BohuTANG) [inline tests/suites/0_stateless/05_ddl/05_0007_ddl_drop_user_v2.sql:14]: It would be better to teardown with `set enable_planner_v2=0`;

Comment (TCeason) [reply tests/suites/0_stateless/05_ddl/05_0007_ddl_drop_user_v2.sql:14]: We has already migrate create user to planner v2. And in 05_0007_ddl_drop_user.sql the test is enable_planner_v2=0. I think if we migrate all statement to planner v2 we can delete old test.

**Comment by TCeason** [comment]
Maybe we can merge this pr?@[BohuTANG](https://github.com/BohuTANG)

**Comment by BohuTANG** [comment]
Conflicting files :/

**Comment by TCeason** [comment]
> Conflicting files :/

Already resolve it. Just wait ci pass.

**Pull request merged**